app/assets/javascripts/components/common/feedback.jsx

```diff
@@ -44,6 +45,19 @@ const Feedback = React.createClass({
     this.hide();
   },
 
+  handleFeedbackInputChange(event) {
+    this.setState({ feedbackInput: event.target.value });
+  },
+
+  handleFeedbackSubmit(event) {
+    const feedback = this.state.feedbackInput;
+    this.setState({ feedbackInput: '' });
+    API.createCustomFeedback(this.props.assignment.id, feedback).then(() => {
+      this.props.feedback[this.props.assignment.id].custom.push(<li>{feedback}</li>);
+    });
```

Comment:
You should move the API call into FeedbackActions, and then dispatch an action that includes the feedback input as data. The action would be something like in alert_actions:

```
export function createCustomFeedback(input) {
  return function (dispatch) {
    dispatch({ type: types.CUSTOM_FEEDBACK_SUBMITTED, data: input }); // handle this in the reducer to add the new feedback into the set of already-received feedback. connect() will then pass in the updated feedback prop and cause the component to re-render.
    return API.createCustomFeedback(input)
      .then(() => (dispatch({ type: types.CUSTOM_FEEDBACK_CREATED }))) // optional, to confirm successful creation of feedback
      .catch(response => (ApiFailAction.fail(response)));
  };
}
```